### PR TITLE
Small 2hr/NM fixes

### PR DESCRIPTION
**Pull request by ymkc**
- Mobs with Manafont active should cast non-stop while effect is up
- Added a check to phOnDespawn to prevent PHs from disappering when having luck on the rng on multi-ph nms

### CODE REVIEW

File: scripts/globals/mobs.lua [hunk 1]
```diff
@@ -59,14 +59,16 @@ function phOnDespawn(ph,phList,chance,cooldown)
         local nm = GetMobByID(nmId);
         if (nm ~= nil) then
             local pop = nm:getLocalVar("pop");
-            if (os.time() > pop and not lotteryPrimed(phList) and math.random(100) <= chance) then
+            local flag = nm:getLocalVar("flag");
```
Comment (TeoTwawki) [inline scripts/globals/mobs.lua]: could we call that something more descriptive?

**Comment by TeoTwawki** [comment]
Small not really important note: since retail changed the old 2hrs to being 1 hour..I've been trying to call them "job special" instead. hence "jobSpec" table. Been naming the local vars like "hpTrigger" or "benedictionTrigger" and such to avoid time references on non time based things. I'll still merge without that it is not a problem, just hoping to get other opinions and get us all on some kind of consensus on what to call stuff in the future and then we can get all the script on the same style later.

File: scripts/globals/mobs.lua (hunk 1 above)
Comment (ymkc) [reply scripts/globals/mobs.lua]: trying hard ... :)

**Comment by ymkc** [comment]
@teotwawki hate to think of var names anyway, so will use whatever will be decided on

File: scripts/globals/mobs.lua [hunk 2]
```diff
@@ -59,14 +59,16 @@ function phOnDespawn(ph,phList,chance,cooldown)
         local nm = GetMobByID(nmId);
         if (nm ~= nil) then
             local pop = nm:getLocalVar("pop");
-            if (os.time() > pop and not lotteryPrimed(phList) and math.random(100) <= chance) then
+            local swapped = nm:getLocalVar("swapped");
```
Comment (wrenffxi) [inline scripts/globals/mobs.lua]: what's the point of this swapped localVar? the lotteryPrimed() function already tells us if a lottery NM is already set to spawn in place of its PH.

Comment (ymkc) [reply scripts/globals/mobs.lua]: oh i see, it's done via nm:getRespawnTime
@wrenffxi totally didn't see it, thanx

**Pull request merged**